File: TravelAgencyFrontend/src/app/types/Reservation/ReservationRequest.ts
import {PaymentStatus} from "./PaymentStatus";
import {PaymentType} from "./PaymentType";
import {CompanionRequest} from "../Companion/CompanionRequest";


export interface ReservationRequest {
  id?: string;
  paymentStatus: PaymentStatus;
  paymentType: PaymentType;
  travelId: string;
  packName: string;
  discount?: number;
  companions?: CompanionRequest[];
  hotelId: string;
  hosting?: string;
  clientId: string;
}

File: TravelAgencyRestApi/src/main/java/com/travelagency/travelagencyapplication/dto/request/ReservationRequest.java
package com.travelagency.travelagencyapplication.dto.request;

import com.travelagency.travelagencyapplication.collection.*;
import com.travelagency.travelagencyapplication.enums.PaymentStatus;
import com.travelagency.travelagencyapplication.enums.PaymentType;
import lombok.*;

import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ReservationRequest {

    private boolean updatedMethod;

    private PaymentStatus paymentStatus;

    private PaymentType paymentType;

    private String travelId;

//    private Set<PackRequest> packs;
    private String packName;

    private String hotelId;

    private String hosting;

    private String clientId;

    private Double discount;

    private Set<String> companionsId;

}
